source/libraries/CompileSolidity.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as solc from 'solc';
+
+class SolidityContractCompiler {
+    private compiler: any = solc;
+    private contractInputDirectoryPath: string;
+    private contractOutputDirectoryPath: string;
+    private contractOutputFileName: string;
+
+    public constructor(contractInputDirectoryPath: string, contractOutputDirectoryPath: string, contractOutputFileName: string) {
+        this.contractInputDirectoryPath = contractInputDirectoryPath;
+        this.contractOutputDirectoryPath = contractOutputDirectoryPath;
+        this.contractOutputFileName = contractOutputFileName;
+    }
+
+    public compileContracts() {
+        // Compile all contracts in this.inputDirectoryPath
+        let inputJSON: any = this.generateCompilerInput();
+        let compilerOutput: any = this.compiler.compile(inputJSON);
+        if (compilerOutput.errors.length > 0) {
+            console.log("The compiler encountered the following warnings/errors:");
+            for (let error of compilerOutput.errors) {
+                console.log(error);
+            }
+            // Do not return automatically in case there are only warnings and no errors
```

Comment:
I think compilation should probably fail on both warnings and errors, so I'm OK with just throwing here.  Also, we may want to consider using a custom Error type that includes the compiler errors and then letting the caller decide how they want to deal with compiler errors.  It _feels_ like this is the wrong place to print the errors (what if the caller wants to write errors to a file, or wants to redirect errors to stderr instead of stdout).

Reply:
Yeah, I agree.  Error handling is something I figured we'd need to discuss further, so I hadn't spent a lot of time on it.